### ISSUE TITLE
rook-ceph-cluster chart uses wrong provisioner for objectStore `StorageClass`

### ISSUE THREAD

**Issue by SPodjasek** [rook/rook#10894]
**Is this a bug report or feature request?**
* Bug Report

This issue was already described, discussed and closed in #10153 - but it still exists, so IMHO changes introduced in 36a93745553f624f92b0dfbe59f8492677635de7 break things now.

**How to reproduce it (minimal and precise):**

1. Deploy operator to default namespace `rook-ceph`
2. Deploy cluster chart to different namespace (ex. `cluster-namespace`) with `cephObjectStores[*].storageClass` enabled
3. Create `ObjectBucketClaim` in `cluster-namespace`
4. Observe Operator logs and look for:
    ```
    I0906 10:54:55.857514       1 controller.go:238]  
      "msg"="unsupported provisioner" 
      "key"="cluster-namespace/ceph-delete-bucket" 
      "got"="rook-ceph.ceph.rook.io/bucket"
    ```
5. After creating new `StorageClass` with provisioner set to: `cluster-namespace.ceph.rook.io/bucket`, and creating new claim with new `storageClassName` bucket is created as expected

**Environment**:
* OS: `Ubuntu 22.04.1 LTS`
* Kernel: `Linux w5 5.15.0-43-generic #46-Ubuntu SMP Tue Jul 12 10:30:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`
* Cloud provider or hardware configuration: `on-premise cluster`
* Rook version: `v1.9.10`
* Operator Helm chart: `rook-ceph-v1.9.10`
* Cluster Helm chart: `rook-ceph-cluster-v1.9.10`
* Storage backend version: `17.2.3`
* Kubernetes version: `v1.24.2`
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): `kubeadm`

**Comment by travisn** [comment]
Did you set the `operatorNamespace` in the chart values? The default for that setting is the same namespace as the object store, but you will need to set it to the other namespace. This setting is picked up by the storage class [here](https://github.com/rook/rook/blob/master/deploy/charts/rook-ceph-cluster/templates/cephobjectstore.yaml#L16).

**Comment by SPodjasek** [comment]
Yes it's set to `rook-ceph`. 
Maybe I've understood the comment above it wrong:
```
# Namespace of the main rook operator
operatorNamespace: rook-ceph
```
so I thought it refers to namespace where [Rook Ceph Operator](https://rook.io/docs/rook/v1.9/Helm-Charts/operator-chart/) is deployed, not the namespace where I want my cluster and resources to be deployed....

**Comment by SPodjasek** [comment]
Looking at the docs further:
> If the operator was installed in a namespace other than rook-ceph, the namespace must be set in the operatorNamespace variable.

So in my case, operator was installed in default namespace `rook-ceph` so the value was set correctly. 

But the operator won't work with provisioner `rook-ceph.ceph.rook.io/bucket` (as set by chart values), only `cluster-namespace.ceph.rook.io/bucket` works.

**Comment by travisn** [comment]
I see, the object provisioner prefix should be the namespace of the object store. @BlaineEXE Is that correct? 

This is different from the rbd, cephfs, or nfs storage classes which use the operator namespace as the prefix.

**Comment by BlaineEXE** [comment]
Yes, I believe the prefix should be the cluster namespace, not the operator namespace. From what I can tell, I think the bucket controller watches for new ceph clusters and then that controller spins up bucket provisioner routines for each cluster.

**Comment by BlaineEXE** [comment]
@SPodjasek would you like to validate the fix in #10904 before it's merged?

**Comment by SPodjasek** [comment]
@BlaineEXE Yes it works after using chart from this PR
